common/headtracker/head_tracker.go

```diff
@@ -349,22 +401,22 @@ func (ht *headTracker[HTH, S, ID, BLOCK_HASH]) calculateLatestFinalized(ctx cont
 			return latestFinalized, fmt.Errorf("failed to get valid latest finalized block")
 		}
 
-		if currentHead.BlockNumber()-latestFinalized.BlockNumber() > int64(ht.htConfig.MaxAllowedFinalityDepth()) {
-			return latestFinalized, fmt.Errorf("gap between latest finalized block (%d) and current head (%d) is too large (> %d)",
-				latestFinalized.BlockNumber(), currentHead.BlockNumber(), ht.htConfig.MaxAllowedFinalityDepth())
+		if ht.config.FinalizedBlockOffset() == 0 {
+			return latestFinalized, nil
 		}
 
-		return latestFinalized, nil
+		finalizedBlockNumber := max(latestFinalized.BlockNumber()-int64(ht.config.FinalizedBlockOffset()), 0)
+		return ht.getHeadAtHeight(ctx, latestFinalized.BlockHash(), finalizedBlockNumber)
```

Comment:
If that's the case shouldn't we add a verify check that `FinalizedBlockOffset` should always be less or equal than `HistoryDepth` ?